### PR TITLE
Fix finally

### PR DESCRIPTION
**Pull request by MBkkt**

### CODE REVIEW

**Review by gnusi — CHANGES_REQUESTED** [review]
Please add `fu2` to https://github.com/iresearch-toolkit/iresearch/blob/master/THIRD_PARTY_README.md

File: core/utils/misc.hpp [hunk 1]
```diff
@@ -33,57 +32,37 @@ namespace iresearch {
...
 // Convenient helper for simulating 'try/catch/finally' semantic
 template<typename Func>
-class finally {
+class [[nodiscard]] Finally {
  public:
   static_assert(std::is_nothrow_invocable_v<Func>);
 
-  explicit finally(Func&& func) : func_{std::forward<Func>(func)} {}
-  ~finally() { func_(); }
+  // If you need some of it, please use absl::MakeCleanup
```
Comment (Dronplane) [inline core/utils/misc.hpp:39]: Why not completely move to absl::Cleanup ?

**Review by Dronplane — APPROVED** [review]
LGTM

File: core/utils/misc.hpp (hunk 1 above)
Comment (MBkkt) [reply core/utils/misc.hpp:39]: We discussed it earlier with @gnusi and for simple expression we want simple (without flag, placement new, etc) abstraction.

But copy/move finally is unsafe (func will be called twice or more time)

**Review by gnusi — APPROVED** [review]

**Pull request merged**